### ISSUE TITLE
DefaultArtifactHandler is useless, and Documentation does not say so!

### ISSUE THREAD

**Issue by cornergraf** [gradle/gradle#3063]
## The Problem
I have a project that builds an RPM through the ant RPM task, obviously this project does not use the java plugin because it does nothing java specific. I wanted to declare the output of my custom RPM task as an artifact, but I kept getting error messages, and only after several attempts did I realize that the culprit was the DefaultArtifactHandler, which apparently doesn't do anything at all. After applying the java plugin things worked, but this really should not be necessary.

## Desired Solutions
1) The DefaultArtifactHandler really should support adding file artifacts.
2) Alternatively, all documentation and examples on adding artifacts should note that this requries a proper ArtifactHandler and how to get one. **NOTE:** This should *definitely not* require the user to apply the java plugin or similar for projects that otherwise have no need for the java plugin.

**Comment by oehme** [comment]
Duplicate of #3064
